src/schema/sources.ts

```diff
@@ -841,6 +870,40 @@ export const resolvers: IResolvers<any, Context> = {
 
       return getSourceById(ctx, info, sourceId);
     },
+    removeMember: async (
+      _,
+      { sourceId, memberId }: RemoveMemberArgs,
+      ctx,
+    ): Promise<GQLEmptyResponse> => {
+      await ensureSourcePermissions(
+        ctx,
+        sourceId,
+        SourcePermissions.RemoveMember,
+      );
+      await ctx.con.transaction(async (manager) => {
+        const repo = manager.getRepository(SourceMember);
+        const [member, loggedUser] = await Promise.all([
+          repo.findOneByOrFail({ sourceId, userId: memberId }),
+          repo.findOneByOrFail({ sourceId, userId: ctx.userId }),
+        ]);
+
+        const memberRank = roleRank[member.role];
+        const loggedUserRank = roleRank[loggedUser.role];
+
+        if (memberRank >= loggedUserRank) {
```

Comment:
Would this mean technically `members` have some kind of access?
(I know we don't have a lower role now)

Reply:
The most basic ones, which are the viewing of Squad, posting, etc. - this is just to make it easy to identify which has a higher hierarchy since we now have more than 2 roles.

Reply:
Understood, just meant to say, technically now users have access to this function.
Not a problem now, but might be if we introduce another role, but maybe that's fine enough for now 🙏

Reply:
Ah. Yes. I misunderstood at first. Unfortunately, we can't use the built-in directives provided by GraphQL as we use the `role` for users differently. So yeah, this is exposed to our users that's why I have written numerous and strict test cases.